### ISSUE TITLE
Failed to connect to OpenSSH_7.2

### ISSUE THREAD

**Issue by xtianus** [int128/gradle-ssh-plugin#259]
I'm using the library with no issues when connecting to a Ubuntu 14.04.4 LTS server.
I can't connect, with the same code, to a Ubuntu 16.04.1 LTS server.

Gradle build:

```
buildscript {
	repositories {
		maven { url "https://plugins.gradle.org/m2/" }
		jcenter()
		mavenCentral()
	}
	dependencies { 
		classpath "org.hidetake:gradle-ssh-plugin:2.4.1"
	}
}

apply plugin: "org.hidetake.ssh" 

remotes {
	deployHost {
		host = "xx.xx.xx.xx
		user = "xxx"
		identity = new File("C:\xxx\mykey.pem")
	}
}

ssh.settings {
	jschLog = true
}

task mytest() << {
	ssh.run {
		session(remotes.deployHost) {
		  execute "ls -l"
		}
	}
}
```

Client log:

```
Connecting to deployHost [xx.xx.xx.xx:22] with {keepAliveSec=60, retryCount=0, retryWaitSec=0, knownHosts=C:\Users\xxx\.ssh\known_hosts, agent=false, identity=C:\xxx\mykey.pem, authentications=[publickey, keyboard-interactive, password], user=xxx}
Using known-hosts file for deployHost: C:\Users\xxx\.ssh\known_hosts
Using public key authentication for deployHost: C:\xxx\mykey.pem
Connecting to xx.xx.xx.xx port 22
Connection established
Remote version string: SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu1
Local version string: SSH-2.0-JSCH-0.1.52
CheckCiphers: aes256-ctr,aes192-ctr,aes128-ctr,aes256-cbc,aes192-cbc,aes128-cbc,3des-ctr,arcfour,arcfour128,arcfour256
aes256-ctr is not available.
aes192-ctr is not available.
aes256-cbc is not available.
aes192-cbc is not available.
CheckKexes: diffie-hellman-group14-sha1,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521
diffie-hellman-group14-sha1 is not available.
CheckSignatures: ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521
SSH_MSG_KEXINIT sent
SSH_MSG_KEXINIT received
kex: server: curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group14-sha1
kex: server: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
kex: server: none,zlib@openssh.com
kex: server: none,zlib@openssh.com
kex: server: 
kex: server: 
kex: client: diffie-hellman-group1-sha1,diffie-hellman-group-exchange-sha1,diffie-hellman-group-exchange-sha256,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521
kex: client: ssh-rsa,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521
kex: client: aes128-ctr,aes128-cbc,3des-ctr,3des-cbc,blowfish-cbc
kex: client: aes128-ctr,aes128-cbc,3des-ctr,3des-cbc,blowfish-cbc
kex: client: hmac-md5,hmac-sha1,hmac-sha2-256,hmac-sha1-96,hmac-md5-96
kex: client: hmac-md5,hmac-sha1,hmac-sha2-256,hmac-sha1-96,hmac-md5-96
kex: client: none
kex: client: none
kex: client: 
kex: client: 
kex: server->client aes128-ctr hmac-sha1 none
kex: client->server aes128-ctr hmac-sha1 none
SSH_MSG_KEX_DH_GEX_REQUEST(1024<1024<1024) sent
expecting SSH_MSG_KEX_DH_GEX_GROUP
Disconnecting from xx.xx.xx.xx port 22
Waiting for pending sessions
Completed all sessions
Finished executing task ':mytest'
:mytest FAILED
 
FAILURE: Build failed with an exception.
* Where:
Build file 'C:xxx\build.gradle' line: 27
 
* What went wrong:
Execution failed for task ':mytest'.
> Session.connect: java.io.IOException: End of IO Stream Read

```

Sever log:

`sshd[4331]: fatal: mm_answer_moduli: bad parameters: 2048 2048 1024`

Running on Windows with jdk1.7.0_79

**Comment by int128** [comment]
Thanks for details. This issue seems fixed in JSch 0.1.53. I will release an new version with the latest JSch.
https://github.com/rundeck/rundeck/issues/1796

**Comment by xtianus** [comment]
Thank you.
While waiting for the release, this fixed the problem:

```
	dependencies { 
		classpath "org.hidetake:gradle-ssh-plugin:2.4.1",
		'com.jcraft:jsch:0.1.53'
	}

```

**Comment by int128** [comment]
Released 2.4.2.
